doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
+
++ Request (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        + password: aSecretPassword (string, required)
+        
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "username": {
+                        "description": "Name of the user that wants to authenticate",
+                        "type": "string"
+                    },
+                    "password": {
+                        "description": "Password for the username in question",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "username",
+                    "password"
+                ],
+                "additionalProperties": false
+            }
+ 
++ Response 200 (application/json)
+    + Body
+
+            {
+                "status": 200,
+                "token": "eyJ...Q2tY="
+            }
+            
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "token": {
+                        "description": "Session token to be used as authentication for further requests.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "token"
+                ],
+                "additionalProperties": false
+            }
+
++ Response 401 (application/json)
+    + Body
+    
+            {
+                "status": 401,
+                "error": "Unauthorized",
+                "message": "You have to supply an username and a password.",
+                "loca": "AUTH_NEED_USERNAME_PASSWORD"
```

Comment:
Can you give me links to json schema or blueprint explanations where I can lookup what `loca` means?